### PR TITLE
[cli] Normalize "src" property in `vc build`

### PR DESCRIPTION
**Pull request by TooTallNate**
This matches the production behavior.

### CODE REVIEW

**Review by styfle — APPROVED** [review]
Thanks!

File: packages/cli/src/commands/build.ts [hunk 1]
```diff
@@ -477,15 +478,33 @@ export default async function main(client: Client): Promise<number> {
...
+  if (!build.use) {
+    throw new NowBuildError({
+      code: `invalid_build_specification`,
+      message: 'Field `use` is missing in build specification',
+      link: 'https://vercel.com/docs/configuration#project/builds',
+      action: 'View Documentation',
+    });
+  }
+
+  let src = normalize(build.src || '**');
```
Comment (pvdz) [inline packages/cli/src/commands/build.ts:488]: I don't know what "the production behavior" is but this will change a few things that may trigger subtle bugs (or fix them, I can't tell)

- resolve `.` and `..`
- replace slashes with their platform native slash
  - if this is `vc build` could this affect windows users? Does our system expect all slashes to be normalized to linux slashes?
- an empty string results in `.` (could hide bugs?)

As per https://nodejs.org/api/path.html#pathnormalizepath

File: packages/cli/src/commands/build.ts [hunk 2]
```diff
@@ -475,17 +476,33 @@ export default async function main(client: Client): Promise<number> {
 }
 
 function expandBuild(files: string[], build: Builder): Builder[] {
-  if (!build.src) return [];
+  if (!build.use) {
```
Comment (pvdz) [inline packages/cli/src/commands/build.ts:479]: This change seems unrelated to the PR :shrug:

**Review by pvdz — APPROVED** [review]
I assume that this indeed matches production and doesn't introduce expected changes :)
